From 4732b62e7cb7a386484f86a3955523bb61a7ec61 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 20 Sep 2024 09:48:36 +0200
Subject: [PATCH 736/841] feat(stm32mp2): forbid upper OTP access with mirror

The upper OTP fuses are not mirrored in bsec_mirror area. Return an
error if such access is tried.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git f78486438df6338ccb45880b1eda19d825d45298 feat(stm32mp2): forbid upper OTP access with mirror]
Change-Id: I8318e79c52eff525585838e0ed199cdb4957e7a1
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/412863
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_otp_mirror.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_otp_mirror.c b/plat/st/stm32mp2/stm32mp2_otp_mirror.c
index 1acd34566..42afa5263 100644
--- a/plat/st/stm32mp2/stm32mp2_otp_mirror.c
+++ b/plat/st/stm32mp2/stm32mp2_otp_mirror.c
@@ -22,6 +22,11 @@ uint32_t otp_mirror_read(uint32_t *val, uint32_t otp)
 {
 	struct otp_mirror *mirror = (struct otp_mirror *)SRAM1_BASE;
 
+	/* Upper OTPs are not mirrored */
+	if (otp >= STM32MP2_UPPER_OTP_START) {
+		return BSEC_NOT_SUPPORTED;
+	}
+
 	/* OTP already mirrored */
 	if ((mirror->magic == OTP_MIRROR_MAGIC) && (mirror->state != BSEC_STATE_INVALID)) {
 		*val = mirror->value[otp];
-- 
2.47.2

